### PR TITLE
Move IsLocalAddress outside of the config package

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Move `IsLocalAddress` outside of the config package. This reduce the dependencies on `pkg/config` when not needed.

This commit also make a module out of pkg/util/system.

### Describe how to test/QA your changes

No code change.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/util/system/go.mod [hunk 1]
```diff
@@ -0,0 +1,32 @@
+module github.com/DataDog/datadog-agent/pkg/util/system
+
+go 1.21
+
+require (
+	github.com/DataDog/datadog-agent/pkg/util/filesystem v0.50.2
+	github.com/DataDog/datadog-agent/pkg/util/log v0.50.2
+	github.com/DataDog/datadog-agent/pkg/util/pointer v0.50.2
+	github.com/DataDog/datadog-agent/pkg/util/testutil v0.50.2
+	github.com/DataDog/datadog-agent/pkg/util/winutil v0.36.1
```
Comment (pgimalac) [inline pkg/util/system/go.mod:10]: **🥜 nitpick:** ‏I think you need to add `replace` directives for those imports (+ `pkg/util/scrubber` which is indirect)

**Review by pgimalac — APPROVED** [review]
LGTM, some replace directives are missing but I think the CI will tell you anyway

**Review by ogaca-dd — APPROVED** [review]

**Review by carlosroman — APPROVED** [review]

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned file !

**Pull request merged**